src/types/receipt.ts

```diff
@@ -0,0 +1,123 @@
+// Types
+import { JsonRpcLog } from "./log.ts";
+
+// Starknet
+import { Event } from "../deps.ts";
+
+// Eth
+import {
+  bigIntToHex,
+  Bloom,
+  generateAddress,
+  hexToBytes,
+  JsonRpcTx,
+} from "../deps.ts";
+import { bytesToHex } from "../deps.ts";
+
+/**
+ * @param transaction A Ethereum transaction.
+ * @param logs A array of Ethereum logs.
+ * @param event The "transaction_executed" event.
+ * @param cumulativeGasUsed The cumulative gas used up to this transaction.
+ * @returns The Ethereum receipt.
+ */
+export function toEthReceipt(
+  { transaction, logs, event, cumulativeGasUsed }: {
+    transaction: JsonRpcTx;
+    logs: JsonRpcLog[];
+    event: Event;
+    cumulativeGasUsed: bigint;
+  },
+): JsonRpcReceipt | null {
+  // Gas used is the last piece of data in the transaction_executed event.
+  // https://github.com/kkrt-labs/kakarot/blob/main/src/kakarot/accounts/eoa/library.cairo
+  const gasUsed = bigIntToHex(BigInt(event.data[event.data.length - 1]));
+  // Status is the second to last piece of data in the transaction_executed event.
+  // https://github.com/kkrt-labs/kakarot/blob/main/src/kakarot/accounts/eoa/library.cairo
+  const status = bigIntToHex(BigInt(event.data[event.data.length - 2]));
+  // If there is no destination, calculate the deployed contract address.
+  const contractAddress = transaction.to === null
+    ? bytesToHex(generateAddress(
+      hexToBytes(transaction.from),
+      hexToBytes(transaction.nonce),
```

Comment:
i think there's a trap in the nonce:
https://github.com/ethereum/execution-specs/blob/559ba54829a5d726ca95fe5701486f7bd303bf7f/src/ethereum/paris/utils/message.py#L79

the nonce is already incremented no?